Modules/_localemodule.c

```diff
@@ -748,12 +745,14 @@ PyInit__locale(void)
     PyModule_AddIntMacro(m, LC_ALL);
     PyModule_AddIntMacro(m, CHAR_MAX);
 
-    Error = PyErr_NewException("locale.Error", NULL, NULL);
-    if (Error == NULL) {
-        Py_DECREF(m);
-        return NULL;
+    _locale_state *state = get_locale_state(m);
+    state->Error = PyErr_NewException("locale.Error", NULL, NULL);
```

Comment:
This `Error` object is never freed. Worse, the only reference to it ends up in in the `_locale` module's globals, from where it can be removed by Python code. With this PR, the following script will crash the interpreter:

```
import _locale
import gc

del _locale.Error
gc.collect()

_locale.bindtextdomain("", "")
```

Please have the C-level module state hold a reference, and add `m_traverse`/`m_clear`/`m_free` to properly release it.

Reply:
Oh, thanks, petr. I don't notice those slots :(.
I will update it later.

Reply:
If I understand clearly, this extension module need add clean slots too.
refs: https://github.com/python/cpython/blob/master/Modules/binascii.c#L1616

Reply:
Hi, petr. I am not sure it's right to return `-1` when checking `state` is NULL in `m_traverse`/`m_clear`/`m_free`